pkg/blobinfocache/memory/memory.go

```diff
@@ -120,24 +120,42 @@ func (mem *cache) RecordDigestCompressorName(blobDigest digest.Digest, compresso
 	mem.compressors[blobDigest] = compressorName
 }
 
-// appendReplacementCandidates creates prioritize.CandidateWithTime values for (transport, scope, digest), and returns the result of appending them to candidates.
-func (mem *cache) appendReplacementCandidates(candidates []prioritize.CandidateWithTime, transport types.ImageTransport, scope types.BICTransportScope, digest digest.Digest, requireCompressionInfo bool) []prioritize.CandidateWithTime {
-	locations := mem.knownLocations[locationKey{transport: transport.Name(), scope: scope, blobDigest: digest}] // nil if not present
-	for l, t := range locations {
-		compressorName, compressorKnown := mem.compressors[digest]
-		if !compressorKnown {
-			if requireCompressionInfo {
-				continue
-			}
-			compressorName = blobinfocache.UnknownCompression
+// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket
+// (which might be nil) with corresponding compression
+// info from compressionBucket (which might be nil), and returns the result of appending them
+// to candidates.
+// v2Output allows including candidates with unknown location, and filters out candidates
+// with unknown compression.
+func (mem *cache) appendReplacementCandidates(candidates []prioritize.CandidateWithTime, transport types.ImageTransport, scope *types.BICTransportScope, digest digest.Digest, v2Output bool) []prioritize.CandidateWithTime {
```

Comment:
https://github.com/containers/image/pull/1645#discussion_r970052118 : `scope` doesn’t need to be a pointer to mark optionality.